app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || AnonymousUser.new
```

Comment:
It does say `Password Reset Requested for #{analytics_user.role} role`. The example above was just for illustration purposes. I don't think we want to log different text. I think using the same text serves the purpose. `Password reset requested for nonexistent user` communicates the event clearly I think.